bundles/org.eclipse.cdt.lsp.editor.ui/src/org/eclipse/cdt/lsp/editor/ui/preference/LspEditorPreferencePage.java

```diff
@@ -37,7 +48,52 @@ public void init(IWorkbench workbench) {
 	@Override
 	public void createFieldEditors() {
 		PreferenceMetadata<Boolean> prefer = LspEditorPreferences.getPreferenceMetadata();
-		addField(new BooleanFieldEditor(prefer.identifer(), prefer.name(), getFieldEditorParent()));
+		var booleanFieldEditor = new BooleanFieldEditor(prefer.identifer(), prefer.name(), getFieldEditorParent());
+		addField(booleanFieldEditor);
+		
+		serverPath = new FileFieldEditor(LspEditorPreferences.SERVER_PATH,
+				LspEditorUiMessages.LspEditorPreferencePage_server_path, getFieldEditorParent());
+		addField(serverPath);
+
+		
+		serverOptions = new MultiLineTextFieldEditor(LspEditorPreferences.SERVER_OPTIONS,
+				LspEditorUiMessages.LspEditorPreferencePage_server_options, getFieldEditorParent());
+		addField(serverOptions);
+	}
+		
+	@Override
+	public boolean performOk() {
+		writeServerPathToProvider(serverPath.getStringValue());
+		writeServerOptionsToProvider(serverOptions.getStringValue());
+		return super.performOk();
+	}
+	
+	private void writeServerPathToProvider(String path) {
+		if (path == null || path.isBlank()) {
+			return;
+		}
+		List<String> commands = LspPlugin.getDefault().getCLanguageServerProvider().getCommands();
+		if (commands != null && !commands.isEmpty()) {
+			commands.set(0, path);
+		} else if (commands == null) {
+			commands = new ArrayList<>();
+			commands.add(path);
+		}
+		LspPlugin.getDefault().getCLanguageServerProvider().setCommands(commands);
+	}
+	
+	private void writeServerOptionsToProvider(String options) {
+		if (options == null || options.isBlank()) {
```

Comment:
Blank options should be allowed. 

At the moment if a user enters blank options the preference store stores the blank value, but this method does not update the arguments passed to the provider.

Reply:
good point